docs/blog/blog.md

```diff
@@ -0,0 +1,221 @@
+
+![Feature interpretation](./interpretation.png "Feature interpretation")
+
+Figure 2: Dominant keywords describing output vector’s dimensions
+
+## Evaluation
+
+For evaluating the performance of these methods, we take a dataset with different employment position titles and their corresponding remuneration. We encode the categorical variable _position title_ using the methods described above. Then, we define a dummy binary classification task of predicting whether or not the employee earns more than 100,000 units. To evaluate the performance of each encoding method, we use the same xgboost classification algorithm. Figure 3 shows the results obtained. Word-based encoders outperform the base N encoders. The hashing encoder performed the worst followed by one-hot, integer, and binary encoder. Similarity encoder performed the best with online gamma poisson factorization (OGPF) as a close second.
+
+![Results](./results.png "Results")
+
+Figure 3: Weighted average f1 scores of encoding schemes
+
+## Application
+
+Next, we are going to see how we can apply gamma poisson factorization in an unsupervised setting.
+
+![Dataset](./dataset.png "Data set")
+
+Figure 4: Data set of system hardware configurations
+
+- **Data:** We use a data set of hardware configuration of user systems. Figure 4 shows the system configurations.
+
+- **Problem:** Our aim is to find configurations that are outliers and rules that determine that these configurations are outliers.
+
+- **Encoding:** We use gamma poisson factorization for the variable _modelname_ since it has a high cardinality. For the rest, we use one-hot encoding since they have a low cardinality.
+
+- **Outlier detection:** To find the outliers, we use the isolation forest algorithm that finds points that are easiest to isolate.
+
+- **Interpretation:** Finally, we train a surrogate decision tree based on the labels given by the isolation forest to find rule paths that lead to the outlier classification.
+
+#### **Interpretation**
+
+![Rulepath](./rulepath.png "Rule path")
+
+Figure 5: Example rule path for detecting a system as an anomaly
+
+##### The tree can be interpreted in the following way:
+
+- The nodes represent feature splitting points.
+- The first line represents a condition used to split that node.
+- The second line specifies the percentage of samples that fall in that node.
+- The third line specifies that, for those samples that fall in that node, what percent belong to class Anomaly (orange color) and what % belongs to class Not Anomaly (Blue color)
+
+##### Rule paths
+
+- One-hot encoded variable: if a node says that vendor_seabios &lt;= 0.5, that means that for all the systems where the vendor is not seabios. Since one-hot encoding means 0 or 1, &lt; 0.5 means 0.
```

Comment:
```suggestion
- One-hot encoded variable: if a node says that `vendor_seabios <= 0.5`, that means that for all the systems where the vendor is not _seabios_. Since one-hot encoding means 0 or 1, &lt; 0.5 means 0.
```